src/rate.rs

```diff
@@ -0,0 +1,210 @@
+use std::convert::TryFrom;
+use std::str::FromStr;
+
+/// Represent a rate. Note this is designed to support Bitcoin/Dai buy and sell rates (Bitcoin being in the range of 10k-100kDai)
+// The rate is equal to integer * 10e-inv_dec_exp
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct Rate {
+    integer: u32,
+    inv_dec_exp: usize,
```

Comment:
I don't fully understand why we need this.

Reply:
Yes, as discussed I am doing _heap floating points_ and I will look into moving to _heap static points_ with predefined precision for each use case.

Reply:
Thanks for stating the obvious, that wasn't clear to me :)